src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -738,6 +740,39 @@ function validate_connection_args(user_input, action) {
     }
 }
 
+////////////////////////////
+//// UPGRADE VALIDATION ////
+///////////////////////////
+
+/**
+ * validate_expected_version checks that expected_version is valid
+ * if it's not valid it throws an error
+ * @param {string} expected_version
+ */
+function validate_expected_version(expected_version) {
+    if (!expected_version) {
+        throw new Error('expected_version flag is required');
+    }
+    if (!version_utils.is_valid_sematic_version(expected_version)) {
+        throw new Error('expected_version must have sematic version structure (major.minor.patch)');
+    }
+    if (check_if_expected_version_newer_that_package_version(expected_version)) {
+        throw new Error(`expected_version cannot be later that the package version ${pkg.version}`);
```

Comment:
what happens if the expected version equals to pkg.version?

Reply:
I do not fail it from the CLI perspective (I saw in the tests that adding it fails parts of the tests).
I had an assumption that there are cases where the upgrade did not finish, and we wanted to rerun it, and didn't want the CLI to block it on the argument.